get_prom_actors_script.py

```diff
@@ -1 +1,130 @@
-# blank file
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Mar 13 17:13:51 2020
+
+@author: meli
+"""
+
+#lirbaries needed to run analysis
+import csv
+from operator import itemgetter
+import networkx as nx
+
+# Openning and reading files
+with open('RQ1ATT.csv', 'r') as nodecsv:
+    nodereader = csv.reader(nodecsv)
+    nodes = [n for n in nodereader][1:]
+
+node_names = [n[0] for n in nodes] # Get a list of only the node names
+
+with open('RQ1EL.csv', 'r') as edgecsv: # Open the file
+    edgereader = csv.reader(edgecsv) # Read the csv
+    edges = [tuple(e[0:3]) for e in edgereader][1:] # Retrieve the data
+
+
+#Creating the Network
+DG = nx.DiGraph()
+DG.add_nodes_from(node_names)
+DG.add_weighted_edges_from(edges)
+
+weight =nx.get_edge_attributes(DG,'weight')
+
+
+# Centrality Measures
+degree_dict = dict(DG.degree(weight=weight.values()))
+betweenness_dict = nx.betweenness_centrality(DG, weight = weight.values(), normalized = False)
+eigenvector_dict = nx.eigenvector_centrality (DG)
+
+# Setting node attributes for centrality measures
+nx.set_node_attributes(DG, degree_dict, 'degree')
+nx.set_node_attributes(DG, betweenness_dict, 'betweenness')
+nx.set_node_attributes(DG, eigenvector_dict, 'eigenvector')
+
+betweenness_dict.keys()
+
+
+
+lst = list(zip(degree_dict.keys(), degree_dict.values(), betweenness_dict.values(), eigenvector_dict.values()))
+
+with open('Centrality_Measures.csv','w') as out:
+    csv_out=csv.writer(out)
+    csv_out.writerow(['Subreddit', 'Degree', 'Betweenness', 'Eigenvector'])
+    for row in lst:
+        csv_out.writerow(row)
+#----------------------------------------------------------------------#
+
+def Get_Promient_Actors_by_cent(cent):
+    if cent == "degree":
+        cent = degree_dict.items()
+    elif cent == "betweenness":
+        cent = betweenness_dict.items()
+    else:
+        cent = eigenvector_dict.items()
+    sorted_cent = sorted(cent, key=itemgetter(1), reverse=True)
+    Prom_Act = sorted_cent[:int(len(sorted_cent) * .20)]
+    Prom_Act_No_0 = [ i for i in Prom_Act if i[1] != 0.0]
+    return Prom_Act_No_0
+
+Betweenness_Prom_Act = Get_Promient_Actors_by_cent("betweenness")
+
+#ask if this can be cleaned up of if there's a better way of doing this
+def Get_Top_Egin_of_Top_Cent(x):
+    res = [[ i for i, j in x ],
+           [ j for i, j in x ]]
+    Prom_Bet_node = list(res[0])
+    lst = []
+    for element in Prom_Bet_node:
+        Y = eigenvector_dict[element]
+        lst.append(Y)
+    Prom_Act_Eigen = {}
+    for key in Prom_Bet_node:
+        for value in lst:
+            Prom_Act_Eigen[key] = value
+            lst.remove(value)
+            break
+    sorted_egin_of_top_bet  = sorted(Prom_Act_Eigen.items(), key=itemgetter(1), reverse=True)
+    Top_Egin_of_Top_Bet = sorted_egin_of_top_bet[:int(len(sorted_egin_of_top_bet) * .20)]
+    return [nodes[0] for nodes in Top_Egin_of_Top_Bet]
```

Comment:
took me awhile to understand what this function was doing but I'm pretty sure you could write it like this:

```suggestion
def get_top_egin(actors):
    egin = [(name, eigenvector_dict[name] for name, _ in actors]
    sorted_egin  = sorted(egin), key=itemgetter(1), reverse=True)
    top_egin = sorted_egin[:int(len(sorted_egin) * .20)]
    return [name for name, _ in top_egin]
```
The underscore is convention for when you have to have a variable but it's not used for anything.

Same comment about making the 20% an argument.

I would also think about your functions more generally. In this case, your variable names imply that you're using the function to go from a list of prominent actors based on betweenness to a list of prominent actors ranked by egin (whatever that is)

But it seems like the function really takes an arbitrary list of tuples, reorders them based on a different value, and returns just the top 20%. So I changed the names to reflect that more general case